cmd/infra/ibmcloud_powervs/create.go

```diff
@@ -202,73 +217,66 @@ func (infra *Infra) setupBaseDomain(option *CreateInfraOptions) (err error) {
 		return fmt.Errorf("unable to get resource controller")
 	}
 
-	var cisInstance *resourcecontrollerv2.ResourceInstance
-	resourceList, _, err := rcv2.ListResourceInstances(&resourcecontrollerv2.ListResourceInstancesOptions{Name: &option.CISInstance})
+	serviceID, servicePlanID, err := getServicePlanID(cisService, cisServicePlan)
+	resourceList, _, err := rcv2.ListResourceInstances(&resourcecontrollerv2.ListResourceInstancesOptions{ResourceID: &serviceID, ResourcePlanID: &servicePlanID})
```

Comment:
as mentioned earlier, lets handle all the plans

Reply:
handled.